src/mlpack/methods/ann/ffn.hpp

```diff
@@ -142,6 +142,16 @@ class FFN
   void Predict(arma::mat& predictors, arma::mat& results);
 
   /**
+   * Predict the responses to a given set of predictors. The responses will
+   * reflect the output of the given output layer as returned by the
+   * output layer function.
+   *
+   * @param predictors Input predictors.
+   * @return Output predictions of responses into.
+   */
+  arma::mat Predict(arma::mat predictors);
+
```

Comment:
But returning a non-const reference to a local variable is an undefined behavior. And I think `NRVO` is fairly reliable unless [conditional return](https://stackoverflow.com/questions/10522585/when-cant-a-compiler-use-rvo-or-nrvo).